flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/internal/BatchTableEnvImpl.scala

```diff
@@ -118,13 +117,12 @@ abstract class BatchTableEnvImpl(
         // translate the Table into a DataSet and provide the type that the TableSink expects.
         val result: DataSet[T] = translate(table)(outputType)
         // Give the DataSet to the TableSink to emit it.
-        batchSink.emitDataSet(shuffleByPartitionFieldsIfNeeded(batchSink, result))
```

Comment:
why deleting these logic?

Reply:
first, The former logic is to add shuffle no matter whether it is a static partition or not.
second, after we remove `getPartitionFields` from `PartitionableTableSink`, here, we can not get the partitions fields. At present, I don't want to optimize dynamic partition shuffle on legacy planner again.